include/utils/types.hpp

```diff
@@ -104,11 +108,39 @@ struct ModuleTierMappings {
         implementations; ///< Mappings for the individual implementations of the module
 };
 
+struct ModuleInfo {
+    struct Paths {
+        std::filesystem::path etc;
+        std::filesystem::path libexec;
+        std::filesystem::path share;
+    };
+
+    std::string name;
+    std::vector<std::string> authors;
+    std::string license;
+    std::string id;
+    Paths paths;
+    bool telemetry_enabled;
+    bool global_errors_enabled;
+    std::optional<Mapping> mapping;
+};
+
+struct TelemetryConfig {
+    int id;
+};
+
 struct Requirement {
-    Requirement(const std::string& requirement_id_, size_t index_);
+    Requirement() = default;
+    Requirement(const std::string& requirement_id_, size_t index_, std::optional<Mapping> mapping = std::nullopt);
     bool operator<(const Requirement& rhs) const;
     std::string id;
-    size_t index;
+    size_t index = 0;
+    std::optional<Mapping> mapping;
+};
+
+struct RequirementConnection {
```

Comment:
What is a `RequirementConnection`?  Maybe there's a better word for this.  Maybe it is a fulfillment?

Reply:
Renamed to Fulfillment as already used by everestpy